src/components/implementation/resmgr/naive/res_info.h

```diff
@@ -0,0 +1,132 @@
+#ifndef RES_INFO_H
+#define RES_INFO_H
+
+#include <consts.h>
+#include <cos_kernel_api.h>
+#include <cos_defkernel_api.h>
+#include <sl.h>
+#include <resmgr.h>
+#include <memmgr.h>
+
+#define RES_INFO_COMP_MAX_SUBTREE 8
+#define RES_INFO_COMP_MAX_THREADS (MAX_NUM_THREADS*RES_INFO_COMP_MAX_SUBTREE)
+
+extern u64_t res_info_schedbmp;
+
+struct res_shmem_glb_info {
+	int free_region_id;
+	int total_pages;
+
+	int npages[MEMMGR_MAX_SHMEM_REGIONS];
+};
+
+struct res_shmem_info {
+	struct cos_compinfo shcinfo;
+
+	vaddr_t shm_addr[MEMMGR_MAX_SHMEM_REGIONS];
+};
+
+struct res_comp_info {
+	spdid_t cid;
+	int thd_used;
+	struct cos_defcompinfo defci;
+	struct res_shmem_info shminfo;
+	struct sl_thd *tinfo[RES_INFO_COMP_MAX_THREADS]; /* including threads from components in subtree. */
+	int initflag;
+	u64_t chbits; /* all child components */
+	u64_t chschbits; /* child components which are also schedulers. */
+
+	struct res_comp_info *parent;
+	int p_thd_iterator; /* iterator for parent to get all threads created by resmgr in this component so far! */
+	thdcap_t p_initthdcap; /* init thread's cap in parent */
+	thdid_t  initthdid; /* init thread's tid */
+};
+
+struct res_comp_info *res_info_comp_init(spdid_t sid, captblcap_t captbl_cap, pgtblcap_t pgtbl_cap, compcap_t compcap,
+					 capid_t cap_frontier, vaddr_t heap_frontier, vaddr_t shared_frontier,
+					 spdid_t par_sid, u64_t ch, u64_t ch_sch);
+
+struct sl_thd *res_info_thd_init(struct res_comp_info *rci, struct sl_thd *t);
+struct sl_thd *res_info_initthd_init(struct res_comp_info *rci, struct sl_thd *t);
+
+struct res_comp_info *res_info_comp_find(spdid_t s);
+struct sl_thd *res_info_thd_find(struct res_comp_info *r, thdid_t t);
+struct sl_thd *res_info_thd_next(struct res_comp_info *r);
+struct sl_thd *res_info_initthd(struct res_comp_info *r);
+unsigned int res_info_count(void);
+void res_info_init(void);
+
+int res_shmem_region_alloc(struct res_shmem_info *rcur, int num_pages);
```

Comment:
would expect this to both return an identifier, and the address.   Should typedef the id.  Might as well just keep `cbuf_t` so that we don't have to change that later.  Without the typedef, it is hard to figure which arguments return values do what.

Reply:
This is an internal API and I have a separate API that returns the address mapped in `resmgr` and an API for address mapped in the client component. 

But I agree to typedefing the ID, will look into using `cbuf_t` for that.